src/core/QCheck.ml

```diff
@@ -334,31 +334,36 @@ module Gen = struct
 
   (* nat splitting *)
 
-  let nat_split2 n st =
-    if (n < 2) then invalid_arg "nat_split2";
+  let pos_split2 n st =
+    if (n < 2) then invalid_arg "pos_split2";
     let n1 = int_range 1 (n - 1) st in
     (n1, n - n1)
 
-  let pos_split2 n st =
+  let nat_split2 n st =
```

Comment:
Looks good, thanks for fixing this.
Should `nat_split2` raise an exception too, for consistency?
```suggestion
  let nat_split2 n st =
  if (n < 0) then invalid_arg "nat_split2";
```
Sure, `int_range` will complain anyway, but this makes the source of the error clearer.

Reply:
My approach so far has been to skip the "obvious" tests that all parameters should be non-negative, and test the "interesting" relations (strictly positive bounds, or order relations between the parameters). Your suggestion does not hurt either, and it sort of reminded me that I didn't include tests for `range_subset`, so I'll add that to the PR in any case.

Reply:
I made the two changes. Thanks!